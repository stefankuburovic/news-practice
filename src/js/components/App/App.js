import React, { useState } from 'react';
import Container from '../../containers/Container';
import HeaderContainer from '../../containers/HeaderContainer/HeaderContainer';
import './App.scss';

function App() {
    const [article, getArticle] = useState({});
    const [country, setCountry] = useState('GB');
    const [category, setActiveCategory] = useState('');
    const [menuItem, setMenuItem] = useState('Top News');

    const setActive = (menuItem) => {
        return setMenuItem(menuItem);
    };

    const setActiveCountry = (country) => {
        return setCountry(country);
    };

    const setActiveArticle = (article) => {
        return getArticle(article);
    };

    const activeMenuItem = menuItem.props !== undefined ? menuItem.props.values.text : menuItem;
    const isArticle = Object.keys(article).length > 0;
    return (
        <div className="app">
            <h1>News</h1>
            <HeaderContainer
                country={country}
                setActive={setActive}
                isArticle={isArticle}
                active={activeMenuItem}
                setActiveArticle={setActiveArticle}
                setActiveCountry={setActiveCountry}
                setActiveCategory={setActiveCategory}
            />
            <Container
                article={article}
                country={country}
                category={category}
                isArticle={isArticle}
                container={activeMenuItem}
                setActiveArticle={setActiveArticle}
                setActiveCategory={setActiveCategory}
            />
        </div>
    );
}

export default App;
