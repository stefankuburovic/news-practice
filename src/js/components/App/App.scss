.app {
  padding: 0;
  display: flex;
  margin: 0 auto;
  max-width: 80%;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}