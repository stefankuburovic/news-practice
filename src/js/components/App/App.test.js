import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const title = 'News';
let wrapped = shallow(<App>{title}</App>);

describe('App', () => {
    it('should render the App Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });

    it('renders the App children', () => {
        expect(wrapped.find('h1').text()).toEqual(title);
    });
});
