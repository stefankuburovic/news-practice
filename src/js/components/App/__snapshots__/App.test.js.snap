// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App Component correctly 1`] = `
<div
  className="app"
>
  <h1>
    News
  </h1>
  <HeaderContainer />
  <Container />
</div>
`;
